contrib/tools/workflowcheck/workflow/checker.go

```diff
@@ -86,16 +88,21 @@ func NewChecker(config Config) *Checker {
 	if config.DebugfFunc == nil {
 		config.DebugfFunc = log.Printf
 	}
+	// Default deterministic wrappers
+	if config.DeterministicWrappers == nil {
+		config.DeterministicWrappers = map[string][]string{"go.temporal.io/sdk/workflow": {"SideEffect"}}
```

Comment:
Some notes:

1. Can we hardcode this one as always safe? (you'll also be able to get rid of the new dependency)
2. Not sure we need deterministic wrappers exposed to config
3. Include `MutableSideEffect`

Reply:
1. Do you mean hardcode path `go.temporal.io/sdk/workflow` inside `determinism` checker? Now this checker does not know about Temporal 🤔 
2. done
3. done

Reply:
Sorry, yeah, I guess they can be in the object. Not sure it should be called "deterministic wrappers" though (we use that term for many of our async APIs in workflows).

I wonder if we can just add `//workflowcheck:ignore` to the side effect and mutable side effect functions in the SDK source. Or if that doesn't work, set them in the default ident refs. And if those don't work with anonymous functions (or function references or whatever), we should probably make them work.

Reply:
- `//workflowcheck:ignore` won't work since the problem is not in `SideEffect` call but in its argument which are analyzed anyway
- deterministic ident refs should not accept nondeterministic calls as arguments. I added an example for this https://github.com/temporalio/sdk-go/pull/1230/files#diff-4f801589f51fc7947c2a5830f4563fbb0c1559d26f04d49d91d896daff9a2b21R66

So, I think we need another mechanism which just skips checking arguments of a predefined function. Maybe `DeterministicWrapper` is not the right name for this mechanism though.

Reply:
Hrmm, true, makes sense. Maybe `AcceptsNonDeterministicParameters` is best.

Question, with this PR, what happens here:

```go
getRandom := func() string { return uuid.NewString() }
workflow.SideEffect(ctx, getRandom)
``

I am thinking that we should not be failing until a function is referenced or invoked instead of just defined.

Reply:
The `determinism` checker is almost stateless now. It just traverses the AST and looks for non-deterministic behaviour only in the current node. So, we don't know what `getRandom` really means inside the `SideEffect` code.

For the same reason we don't fail in the following example
```
a := time.Now
a()
```

I would suggest creating another issue for this particular problem.

Reply:
Hrmm, yeah, we don't set whether a variable carries non-deterministic. My concern is not about the function reference than the local variable. I will open another issue.

Reply:
Could you merge this please?